### ISSUE TITLE
Correction for color blue in LDFC spectrograms

### ISSUE THREAD

**Issue by towsey** [QutEcoacoustics/audio-analysis#274]
## Describe the bug
The drawing of LDFC spectrograms currently contains a colour blue correction because the blue in an LDFC spectrogram is not as "visible" or as "intense" as the red and the green. The correction occurs in the method < public static Image DrawRgbColourMatrix(double[,] redM, double[,] grnM, double[,] bluM, bool doReverseColour) > in the class LDSpectrogramRGB.cs, line 921 (as of 24/11/2019). 

## Expected behavior 
The colour correction was intended to enhance the blue colour when the red and green indices both had low values (< 0.1). THis was done by adding additional green to create a "more visible" cyan colour. The correction was ad-hoc, i.e. "a hack".

## Diagnostics
Anthony has demonstrated that the current colour correction produces a cyan halo around some "blue" acoustic events which is clearly artefactual and not consistent with what can be seen in the grey scale spectrograms.

### Instructions
I am intending to change the colour correction in the hope that it will produce a more sensible colour effect.

## Screenshots
Anthony has an LDFC spectrogram showing the problem.

**Comment by atruskie** [comment]
The effect we saw:
![image](https://user-images.githubusercontent.com/1452968/69505009-f8053280-0f72-11ea-88c7-2c4fc3670c23.png)

At 38 s/px:

![image](https://user-images.githubusercontent.com/1452968/69505012-05bab800-0f73-11ea-87db-3097ca64fcc1.png)

At 9.65 s/px:

![image](https://user-images.githubusercontent.com/1452968/69505024-2256f000-0f73-11ea-9ccc-c12071398136.png)

https://data.acousticobservatory.org/visualize?siteIds=253,254&extent0=1564476980210&extent1=1564495821019&lane=254

**Comment by towsey** [comment]
Concerning the following comment from Anthony: I have been doing some reading on colour spaces and have decided that to do it properly would take more effort than I have the time. I have reduced the effect of the "correction". I do not have the ability to test on a range of images where the "correction" can be stress tested. I have also changed the normalisation constants for EVN and CVR which are typically rendered in blue.  

Anthony's comment:
We should avoid applying this effect based on on a threshold. We'd suggest some kind of curve be applied to the blue channel to boost it.

Additionally, if you want to boost saturation or brightness you definitely need to use an HSL or HSV model. The RGB colour space is not linear and simple manipulations of it will always produce sub-optimal results.

The ImageSharp library (already references) has HSL, HSV and YCbCr colour space structures we can use to manipulate these colours: https://github.com/SixLabors/ImageSharp/tree/master/src/ImageSharp/ColorSpaces

—
